[Cocoa] Drop configurations for 2020-aligned OSes

litherum opened the pull request:
#### f57baa5fc09ed49ccb1b4bba4f818e9970e460f3
<pre>
[Cocoa] Drop configurations for 2020-aligned OSes
<a href="https://bugs.webkit.org/show_bug.cgi?id=258422">https://bugs.webkit.org/show_bug.cgi?id=258422</a>
rdar://111187791

Reviewed by Alexey Proskuryakov.

All supported Cocoa OSes are later than the 2020-aligned ones.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:

Canonical link: <a href="https://commits.webkit.org/265441@main">https://commits.webkit.org/265441@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f6558c642791439871738d43839bec4bbbf646a5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10903 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11116 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11429 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12550 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/10426 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13493 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11097 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13343 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11062 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11965 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9186 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12954 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9258 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9849 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/17080 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/9228 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10329 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10001 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13235 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10338 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/10445 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8532 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11012 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9617 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3011 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2617 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13887 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11321 "Built successfully") | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/10305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2776 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

litherum commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -347,10 +348,7 @@
-    || (PLATFORM(WATCHOS) && !PLATFORM(IOS_FAMILY_SIMULATOR) && __WATCH_OS_VERSION_MAX_ALLOWED >= 70500) \
-    || (PLATFORM(APPLETV) && !PLATFORM(IOS_FAMILY_SIMULATOR) && __TV_OS_VERSION_MAX_ALLOWED >= 140500) \
-    || (PLATFORM(VISION) && !PLATFORM(IOS_FAMILY_SIMULATOR))
+    || (PLATFORM(IOS_FAMILY) && !PLATFORM(IOS_FAMILY_SIMULATOR))
```
This is just `!PLATFORM(IOS_FAMILY_SIMULATOR)`

litherum commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -410,10 +408,7 @@
-    || (PLATFORM(WATCHOS) && !PLATFORM(IOS_FAMILY_SIMULATOR) && __WATCH_OS_VERSION_MAX_ALLOWED >= 70400) \
-    || (PLATFORM(APPLETV) && !PLATFORM(IOS_FAMILY_SIMULATOR) && __TV_OS_VERSION_MAX_ALLOWED >= 140500) \
-    || (PLATFORM(VISION) && !PLATFORM(IOS_FAMILY_SIMULATOR))
+    || (PLATFORM(IOS_FAMILY) && !PLATFORM(IOS_FAMILY_SIMULATOR))
```
Ditto

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -139,7 +139,7 @@
 #endif // !PLATFORM(MAC) && !PLATFORM(MACCATALYST) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
 
-#if !defined(ENABLE_APPLE_PAY_SESSION_V11) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110300) || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140500) || PLATFORM(VISION))
+#if !defined(ENABLE_APPLE_PAY_SESSION_V11) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110300) || PLATFORM(IOS) || PLATFORM(VISION))
```
I think macOS here is also 2020, and either way, it's best to not touch this line multiple times.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -445,7 +446,7 @@
 #endif
 
-#if !defined(ENABLE_MEDIA_USAGE) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 140000))
+#if !defined(ENABLE_MEDIA_USAGE) && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) || PLATFORM(MACCATALYST))
```
Ditto, isn't this macOS from the same year?

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -838,15 +830,16 @@
 #define HAVE_APPLE_LOW_POWER_MODE_SUPPORT 1
 #endif
 
+// FIXME: Why no macCatalyst?
```
I think that `PLATFORM(MAC)` includes macCatalyst. So this comment isn't correct, and the below is just `PLATFORM(COCOA)`. Also, we can probably remove `HAVE_LIBXSLT_FIX_FOR_RADAR_71864140` completely now, relying on the fix unconditionally.

litherum replied:
I'll remove it in a follow-up patch (I'm aiming for these cleanups to not have any behavior change).

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -869,7 +862,7 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110200) \
```
Same here, isn't this macOS 2020 too?

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -955,9 +948,7 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000 && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300) \
```
Same, and this collapses to `PLATFORM(COCOA)` I think.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1067,12 +1058,8 @@
-    || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 140500)) \
-    || PLATFORM(VISION)
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MAX_ALLOWED >= 110300) \
+    || PLATFORM(IOS_FAMILY)
```
Same.

aproskuryakov commented on Source/WTF/wtf/PlatformUse.h:
```diff
@@ -297,10 +297,10 @@
 #endif
 
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) \
```
Same, I think this also collapses to COCOA.

litherum replied:
The lines below do `&& PLATFORM(IOS_FAMILY_SIMULATOR)` so I don't think that's the same as `COCOA`.

litherum committed:
[Cocoa] Drop configurations for 2020-aligned OSes
https://bugs.webkit.org/show_bug.cgi?id=258422
rdar://111187791

Reviewed by Alexey Proskuryakov.

All supported Cocoa OSes are later than the 2020-aligned ones.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WTF/wtf/PlatformUse.h:

Canonical link: https://commits.webkit.org/265441@main
